### ISSUE TITLE
Handle diffing of replace-with-self lists

### ISSUE THREAD

**Issue by justinbmeyer** [canjs/can-view-live#10]
If someone does:

```
var a = new DefineMap({name: "J"})
var b = new DefineMap({name: "K"});

var list = new DefineList([a,b]);

list.replace([a,b]);
```

this will cause an add and remove event that will cause the entire list to re-render.

It would be nice if we could somehow catch this and use the diff to handle this better.

I think we should probably do this in `list.replace` no matter what.

**Comment by justinbmeyer** [comment]
cc @ilyavf

**Comment by phillipskevin** [comment]
The case where the items in the list are exactly the same is covered by [can-define/list/list splice](https://github.com/canjs/can-define/blob/master/list/list.js#L353-L367) to not trigger any `add` or `remove` events. @ilyavf maybe you can give some more info on what the list looks like in your case.

**Comment by justinbmeyer** [comment]
I just saw this issue with sorting.  I am going to add that it would be nice if the `diff` could tell if the same items are in the list or not.  This can be done in `O(n)`.  We could then shuffle around the DOM, re-using the same DOM nodes.

For this to work, we'd have to wait for the end of a batch ... see what events have been dispatched and do the right thing.

I think I've tried something like this.  I'm not sure what the problem was.
